pkg/devfile/adapters/docker/component/utils.go

```diff
@@ -165,3 +174,49 @@ func (a Adapter) generateAndGetContainerConfig(componentName string, comp versio
 	containerConfig := a.Client.GenerateContainerConfig(*comp.Image, comp.Command, comp.Args, envVars, containerLabels)
 	return containerConfig
 }
+
+func (a Adapter) generateAndGetHostConfig() (container.HostConfig, error) {
+	portMap, err := getPortMap()
+	if err != nil {
+		return container.HostConfig{}, err
+	}
+	hostConfig := container.HostConfig{}
+	if len(portMap) > 0 {
+		hostConfig = a.Client.GenerateHostConfig(false, false, portMap)
+	}
+	return hostConfig, nil
+}
+
+func getPortMap() (nat.PortMap, error) {
+	dir, err := os.Getwd()
+	if err != nil {
+		return nat.PortMap{}, err
+	}
+	envInfo, err := envinfo.NewEnvSpecificInfo(dir)
+	if err != nil {
+		return nat.PortMap{}, err
+	}
+	urlArr := envInfo.GetURL()
+	if len(urlArr) > 0 {
+		portmap := nat.PortMap{}
+		for _, element := range urlArr {
+			if element.ExposedPort > 0 {
+				port, err := nat.NewPort("tcp", strconv.Itoa(element.Port))
+				if err != nil {
+					return nat.PortMap{}, err
+				}
+				portmap[port] = []nat.PortBinding{
+					nat.PortBinding{
+						HostIP:   "127.0.0.1",
+						HostPort: strconv.Itoa(element.ExposedPort),
+					},
+				}
+				log.Info("\nApplying URL configuration")
+				log.Successf("URL https://127.0.0.1:%v created\n", element.ExposedPort)
```

Comment:
We shouldn't hard code https here in case a non-secure port was exposed.

Reply:
How can we tell if it's http or https then?

Reply:
Hmm, good point. Let's leave the scheme out for now then.